src/state.ts

```diff
@@ -14,32 +15,31 @@ export type Events =
       data: DataResponse;
     }
   | {
-      type: "search set";
+      type: "search updated";
       search: string;
     }
-  | {
-      type: "country set";
-      selectedCountry: Country | undefined;
-    };
 
 export const initialState = (): State => ({
-  data: {
-    Global: undefined,
-    Countries: [],
-  },
-  search: "United Kingdom",
+  countries: [],
+  global: undefined,
+  search: "",
   selectedCountry: undefined,
 });
 
 export const reducer: React.Reducer<State, Events> = (state, event) =>
   produce(state, (draft) => {
-    switch (event?.type) {
+    switch (event.type) {
       case "data fetched": {
-        draft.data = event.data;
+        draft.countries = event.data.Countries
+        draft.global = event.data.Global
```

Comment:
It's decision making like this that just doesn't come to mind for me, I don't know why I was keeping all the state in data. I don't ask myself those questions when I'm coding, only once you've made the edit